File: backend/app/models/ai.py
from datetime import datetime
from typing import Optional

from sqlalchemy import DateTime, ForeignKey, JSON, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class AiConversation(Base):
    __tablename__ = "ai_conversations"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"))
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

    user = relationship("User")
    messages = relationship("AiMessage", back_populates="conversation")


class AiMessage(Base):
    __tablename__ = "ai_messages"

    id: Mapped[int] = mapped_column(primary_key=True)
    conversation_id: Mapped[int] = mapped_column(ForeignKey("ai_conversations.id"))
    role: Mapped[str] = mapped_column(String(20))  # user | assistant
    content: Mapped[str] = mapped_column(String(10000))
    actions_taken: Mapped[Optional[dict]] = mapped_column(JSON)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

    conversation = relationship("AiConversation", back_populates="messages")

File: frontend/src/api/ai.ts
import client from "./client";
import type { AiMessage } from "../types";

interface AiConversation {
  id: number;
  user_id: number;
  created_at: string;
}

export async function getConversations(): Promise<AiConversation[]> {
  const response = await client.get<AiConversation[]>("/ai/conversations");
  return response.data;
}

export async function getMessages(conversationId: number): Promise<AiMessage[]> {
  const response = await client.get<AiMessage[]>(`/ai/conversations/${conversationId}/messages`);
  return response.data;
}
